validator/accounts/v2/new.go

```
@@ -0,0 +1,265 @@
+	var keymanager v2keymanager.IKeymanager
+	var err error
+	if isNewWallet {
+		switch wallet.KeymanagerKind() {
+		case v2keymanager.Direct:
+			keymanager = direct.NewKeymanager(ctx, wallet, direct.DefaultConfig())
+		case v2keymanager.Derived:
+			log.Fatal("Derived keymanagers are unimplemented, work in progress")
+		case v2keymanager.Remote:
+			log.Fatal("Remote keymanagers are unimplemented, work in progress")
+		default:
+			log.Fatal("Keymanager type must be specified")
+		}
+		keymanagerConfig, err := keymanager.MarshalConfigFile(ctx)
+		if err != nil {
+			log.Fatalf("Could not marshal keymanager config file: %v", err)
+		}
+		if err := wallet.WriteKeymanagerConfigToDisk(ctx, keymanagerConfig); err != nil {
+			log.Fatalf("Could not write keymanager config file to disk: %v", err)
+		}
+		return keymanager
+	}
+	switch wallet.KeymanagerKind() {
+	case v2keymanager.Direct:
+		keymanager, err = direct.NewKeymanagerFromConfigFile(ctx, wallet)
+		if err != nil {
+			log.Fatal(err)
+		}
+	case v2keymanager.Derived:
+		log.Fatal("Derived keymanagers are unimplemented, work in progress")
+	case v2keymanager.Remote:
+		log.Fatal("Remote keymanagers are unimplemented, work in progress")
+	default:
+		log.Fatal("Keymanager kind must be specified")
+	}
+	return keymanager
+}
+
+// Check if a user has an existing wallet at the specified path.
+func hasWalletDir(walletPath string) (bool, error) {
+	_, err := os.Stat(walletPath)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return true, err
+}
+
+func inputWalletDir(cliCtx *cli.Context) string {
+	datadir := cliCtx.String(flags.WalletDirFlag.Name)
+	prompt := promptui.Prompt{
+		Label:    "Enter a wallet directory",
+		Validate: validateDirectoryPath,
+		Default:  datadir,
+	}
+	walletPath, err := prompt.Run()
+	if err != nil {
+		log.Fatalf("Could not determine wallet directory: %v", formatPromptError(err))
```

maybe fatal calls should be handled by the caller of this function ? instead we can just return an error here